src/workerd/api/tests/pipe-streams-test.js

```diff
@@ -173,6 +173,691 @@ export const pipeThroughJsToJs = {
+export const pipeToInternalToJsError = {
+  async test() {
+    const enc = new TextEncoder();
+
+    const { readable, writable } = new IdentityTransformStream();
+
+    const ws = new WritableStream({
+      write(chunk) {},
+    });
+
+    const pipe = readable.pipeTo(ws);
+
+    writable.abort(new Error('boom'));
+
+    await rejects(pipe, { message: 'boom' });
+
+    const writer = ws.getWriter();
+    await rejects(writer.write('hello'), { message: 'boom' });
+  },
+};
+
+// Test pipeTo error in internal readable does not abort JS writable when preventAbort = true
+export const pipeToInternalToJsErrorPrevent = {
+  async test() {
+    const { readable, writable } = new IdentityTransformStream();
+
+    const ws = new WritableStream({
+      write(chunk) {},
+    });
+
+    const pipe = readable.pipeTo(ws, { preventAbort: true });
+
+    writable.abort(new Error('boom'));
+
+    await rejects(pipe, { message: 'boom' });
+
+    ok(!ws.locked);
+
+    const writer = ws.getWriter();
+    await writer.write('hello');
+  },
+};
+
+// Test pipeTo closing internal readable closes JS writable when preventClose = false
+export const pipeToInternalToJsClose = {
+  async test() {
+    const { readable, writable } = new IdentityTransformStream();
+
+    const ws = new WritableStream({});
+
+    const pipe = readable.pipeTo(ws);
+
+    writable.close();
+
+    await scheduler.wait(1);
+
+    const writer = ws.getWriter();
+    await rejects(writer.write('hello'), {
+      message: 'This WritableStream has been closed.',
+    });
+  },
+};
+
+// Test pipeTo closing internal readable does not close JS writable when preventClose = true
+export const pipeToInternalToJsClosePrevent = {
+  async test() {
+    const { readable, writable } = new IdentityTransformStream();
+
+    const ws = new WritableStream({});
+
+    const pipe = readable.pipeTo(ws, { preventClose: true });
+
+    writable.close();
+
+    await scheduler.wait(1);
+
+    const writer = ws.getWriter();
+    await writer.write('hello');
+  },
+};
+
+// Test simple pipeTo JS-to-JS
+export const pipeToJsToJsSimple = {
+  async test() {
+    const chunks = [1, 2, 3];
+    const output = [];
+    const readable = new ReadableStream({
+      async pull(c) {
+        c.enqueue(chunks.shift());
+        if (chunks.length === 0) c.close();
+      },
+    });
+    const writable = new WritableStream({
+      write(chunk) {
+        output.push(chunk);
+      },
+    });
+
+    await readable.pipeTo(writable);
+
+    strictEqual(output[0], 1);
+    strictEqual(output[1], 2);
+    strictEqual(output[2], 3);
+  },
+};
+
+// Test pipeTo error in JS readable aborts JS writable when preventAbort = false
+export const pipeToJsToJsErrorReadable = {
+  async test() {
+    let abortCalled = false;
+    const readable = new ReadableStream({
+      async pull() {
+        throw new Error('boom');
+      },
+    });
+    const writable = new WritableStream({
+      abort(reason) {
+        abortCalled = true;
```

Comment:
Nit: We should use node:test mock more frequently in tests like this: mockFn.mock.callCount() > 0